source/math/FloatMatrix.ooc

```diff
@@ -123,12 +126,13 @@ FloatMatrix : cover {
 	// Tranpose matrix. Creates a new matrix being the transpose of the current matrix.
 	// </summary>
 	// <returns>Return current matrix tranposed.</returns>
-	transpose: func@ -> This {
+	transpose: func -> This {
 		result := This new(this dimensions swap())
 		for (y in 0 .. this height)
 			for (x in 0 .. this width)
 				result elements[y + x * this height] = this elements[x + y * this width]
-		result
+		this free(Owner Callee)
```

Comment:
So in the event that the owner is `Callee`, and the matrix is square, we don't need to allocate memory for a new one but can transpose in-place?